src/btree/bt_walk.c

```diff
@@ -624,3 +625,337 @@ __wti_tree_walk_skip(WT_SESSION_IMPL *session, WT_REF **refp, uint64_t *skipleaf
 
     return (0);
 }
+
+/*
+ * !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ *
+ * Below is a set of functions for computing the normalized position of a page and restoring a page
+ * from its normalized position. These functions are used by the eviction server for the sake of
+ * not holding the hazard pointer for longer than necessary. Another user is "partition cursor".
+ *
+ * Normalized position is a number in the range of 0 .. 1 that represents a page's position across
+ * all pages. It's primary design goal is to be cheap rather than precise. It works best when the
```

Comment:
Its primary design goal

Reply:
There are multiple, actually. Eviction observability is one more.